CeeFM/Decideify
# Books.js Algorithm and Browser Rendering Finalization (pt 1)

**Description**
This pull request is the first in a series that work on finalizing V1 of our suggestion algorithm, browser rendering, and UI/UX functionality for the Books hub / books.js component. We have our genre dropdown menu working now, and submitting it will both manipulate state and data on the page to prepare for a random book suggestion, as well as toggle visibility of buttons the user will need to

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- On the books.js page:
  - You can choose a genre / type for the piece of content from a dropdown menu on each content hub page, and then test that submission and see two things happen
    - 1) the "CategoryId" property of the "suggestion" state variable should be updated to the Id of the option from the "Category" database table for the option you chose
    - 2) the entire Category object that you selected should also be printed to the console after being saved to another state variable named userCategory  
  - After you submit the genre / type form, the form should disappear and be replaced with an image of a treasure chest that says YOUR NEW FAVORITE BOOK on it, as well as buttons to show a book suggestion, and save that suggestion to your database. The save button's visibility at this point will be adjusted in a near-future pull request, as having the option to save something before you see it doesn't really make sense.
  - Clicking the Show Me A Book button should print a random book to the console - if you selected a genre from the dropdown menu on the first page, the random book will be of that type / genre. if you didn't select a genre, it will return a random book of any type/genre.
  - Save suggestion is not wired up to this new book suggestion algorithm just yet though, so it will save a fully random selection, not one based on user input at all even if it was input.

Checklist
